Classes/Telemetry/BITChannel.m

```diff
@@ -192,16 +192,32 @@ void bit_appendStringToSafeJsonStream(NSString *string, char **jsonString) {
   if (string.length == 0) { return; }
   
   char *new_string = NULL;
+  char *prev_jsonString = *jsonString;
+
   // Concatenate old string with new JSON string and add a comma.
   asprintf(&new_string, "%s%.*s\n", *jsonString, (int)MIN(string.length, (NSUInteger)INT_MAX), string.UTF8String);
-  free(*jsonString);
-  *jsonString = new_string;
+
+  // Compare prev_jsonString and *jsonString, if they point to one address then function sets *jsonString to new_string
+  if(OSAtomicCompareAndSwapPtr(prev_jsonString,new_string,(void*)jsonString)) {
+
+    // *jsonString has not been changed, we remove a previous value
+    free(prev_jsonString);
+  } else {
+
+    // *jsonString has been changed
+    free(new_string);
+  }
```

Comment:
This is still unsafe - the `asprintf()` call is unprotected and may access a `*jsonString` which has been freed by another thread. It should use `prev_jsonString` instead, as this will not be freed until the CAS has verified that the value hasn't changed.